### PR TITLE
Support C8S for nginx-container

### PR DESCRIPTION
**Pull request by phracek**
This pull request adds support CentOS Stream 8 for nginx-container.
It also updates README.md files and build-and-push action.

### CODE REVIEW

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test-openshift]

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -44,3 +61,4 @@ jobs:
           docker_context: ${{ matrix.dockerfile_path }}
           dockerfile_path: ${{ matrix.dockerfile_path }}
           tag: ${{ matrix.tag }}
+          suffix: ${{ matrix.suffix }}
```
Comment (zmiklank) [inline .github/workflows/build-and-push.yml:64]:
```suggestion
          suffix: ${{ matrix.tag }}
```

File: .github/workflows/build-and-push.yml [hunk 2]
```diff
@@ -19,22 +19,39 @@ jobs:
...
             quayio_token: "QUAY_IMAGE_BUILDER_TOKEN"
             tag: "centos7"
+            suffix: "centos7"
           - dockerfile_path: "1.20"
             dockerfile: "Dockerfile.c9s"
             registry_namespace: "sclorg"
             quayio_username: "QUAY_IMAGE_SCLORG_BUILDER_USERNAME"
             quayio_token: "QUAY_IMAGE_SCLORG_BUILDER_TOKEN"
             tag: "c9s"
+            suffix: "c8s"
```
Comment (zmiklank) [inline .github/workflows/build-and-push.yml:36]:
```suggestion
            suffix: "c9s"
```
But I would rather won't state suffix here at all, as it seems, that always stays that suffix=tag.

File: 1.20/Dockerfile.c8s [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    mkdir -p ${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start && \
+    mkdir -p ${NGINX_LOG_PATH} && \
+    mkdir -p ${NGINX_PERL_MODULE_PATH} && \
+    chmod -R a+rwx ${NGINX_APP_ROOT}/etc && \
+    chmod -R a+rwx /var/lib/nginx && \
+    chmod -R a+rwx ${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start && \
+    chown -R 1001:0 ${NGINX_APP_ROOT} && \
+    chown -R 1001:0 /var/lib/nginx && \
+    chown -R 1001:0 ${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start && \
+    # FIXME: Not sure if this is safe to do, just a hack to make the image work
```
Comment (zmiklank) [inline 1.20/Dockerfile.c8s:69]: Shouldn't we address this further?

**Review by zmiklank — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-push.yml (hunk 1 above)
Comment (phracek) [reply .github/workflows/build-and-push.yml:64]: Please not, when we will support Fedora build-and-push, the `tag` will not be used.

Comment (phracek) [reply .github/workflows/build-and-push.yml:64]: Or better say, the suffix will not be used at all.

File: 1.20/Dockerfile.c8s (hunk 3 above)
Comment (phracek) [reply 1.20/Dockerfile.c8s:69]: Good question. I don't not know who add this sentence, but it does not work without it.

Comment (zmiklank) [reply 1.20/Dockerfile.c8s:69]: An issue exists for this: https://github.com/sclorg/nginx-container/issues/166

Comment (phracek) [reply 1.20/Dockerfile.c8s:69]: Addressed here: https://github.com/sclorg/nginx-container/pull/187

Comment (phracek) [reply 1.20/Dockerfile.c8s:69]: This should be addressed by the other pull request and not by this one.

Comment (zmiklank) [reply 1.20/Dockerfile.c8s:69]: Agreed.

**Review by zmiklank — COMMENTED** [review]
Please rename this commit: [Update .github/workflows/build-and-push.yml](https://github.com/sclorg/nginx-container/pull/185/commits/c4498dbf4f67d4c4a7336c2de5dcf4e0b39ea628) or sqash it with some other before merging.

Else LGTM.

**Comment by phracek** [comment]
[test-all]

**Pull request merged**